### DISCUSSION TITLE
How to obtain link prediction for a single node in the bipartite_sage_unsup.py example?

### DISCUSSION THREAD

**Discussion by tirzak** [pyg-team/pytorch_geometric#6792] [Q&A] [answered]
Hello,
I am a beginner; I am having a bit of trouble understanding the [bipartite_sage_unsup.py](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/bipartite_sage_unsup.py) example. How would someone obtain link prediction for a single user node after the model has been trained? 

Thank you for your time, and thank you for making such a useful library.

**Comment by rusty1s** [comment, marked as answer]
What you can do is run your GNN as normal, and define the edges you are trying to predict as part of `edge_label_index`. So if you want to make predictions for a single user but all movies, that would be
```python
row = torch.tensor([user_i] * num_movies)
col = torch.arange(num_movies)
edge_label_index = torch.stack([row, col], dim=0)

**Reply by tirzak** [reply]
Thank you very much
